mealplanner-ui/src/pages/Meals/Meal.tsx

```diff
@@ -322,51 +310,44 @@ export const Meal = () => {
                     </tr>
                   </thead>
                   <tbody>
-                    {arrayOfIngredientsWithoutSubstituteIngredients?.map((ingredient) => {
+                    {allIngredients?.map((ingredient) => {
                       return (
-                        <tr key={ingredient.id}>
+                        <tr key={ingredient.rowId}>
                           <td>
-                            {ingredient.name}
-                            {ingredient.substituteIngredient ? (
+                            {ingredient.substituteIngredientId === null && ingredient.name}
+                            {ingredient.substituteIngredientId && (
                               <>
-                                <br />
                                 <span style={{ fontStyle: "italic", marginLeft: "0.5rem" }}>
-                                  Substitute: {ingredient.substituteIngredient.name}
+                                  Substitute: {ingredient?.name}
                                 </span>
                                 <br />
                                 <span style={{ fontStyle: "italic", marginLeft: "0.5rem" }}>
-                                  Reason:{" "}
+                                  Reason:
                                   {ingredient.substituteReason
                                     ? ingredient.substituteReason
                                     : "Not specified"}
                                 </span>
                               </>
-                            ) : (
-                              <></>
                             )}
                           </td>
                           <td style={{ textAlign: "center", verticalAlign: "top" }}>
-                            {ingredient.quantity}
-                            {ingredient.substituteIngredient ? (
+                            {ingredient.substituteIngredientId === null && ingredient.quantity}
```

Comment:
Why do we need this check for ingredient.quantity. Quantity exists irrespective of whether it is substitute Ingredient isnt it?

Reply:
I understand because it is in a different line